eng/illink.targets

```diff
@@ -73,6 +74,8 @@
                             Include="$(ILLinkSuppressionsXmlFile)" />
     <ILLinkSuppressionsXmls Condition="Exists('$(ILLinkSuppressionsConfigurationSpecificXmlFile)')"
                             Include="$(ILLinkSuppressionsConfigurationSpecificXmlFile)" />
+    <ILLinkSuppressionsXmls Condition="Exists('$(ILLinkSuppressionsLibraryBuildXmlFile)')"
```

Comment:
Will this collide with libraries which already define a LibraryBuild.xml?

Reply:
I'm not sure what you mean. No other libraries already define a LibraryBuild.xml. Just CoreLib, but all CoreLib ILLink files are handled specially because it doesn't get binplaced.

Reply:
This is what I meant:
```
C:\Users\joperezr\git\runtime>dir /s /b *LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\coreclr\System.Private.CoreLib\src\ILLink\ILLink.Suppressions.LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.ComponentModel.TypeConverter\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Diagnostics.StackTrace\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Private.CoreLib\src\ILLink\ILLink.Suppressions.LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Private.CoreLib\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Private.Xml\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Private.Xml.Linq\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Security.Claims\src\ILLink\ILLinkTrim_LibraryBuild.xml
C:\Users\joperezr\git\runtime\src\libraries\System.Security.Principal.Windows\src\ILLink\ILLinkTrim_LibraryBuild.xml
```

So corelib is one, but I found others that had a LibraryBuild as well. That said, seems like I was getting confused as those doesn't seem to be suppressions, but instead just the old Linker files that would just preserve types/members.

Reply:
`ILLink.Suppressions.LibraryBuild.xml` is not the same as `ILLinkTrim_LibraryBuild.xml`. The Suppressions file suppresses ILLink warnings. The `ILLinkTrim` files tell the linker what to preserve unconditionally. The `LibraryBuild` part of each means the same thing though - "this file is only used during the build of the library, not in the user's application".

I've been meaning to rename the old name `ILLinkTrim.xml` to `ILLink.Descriptors.xml` (and subsequently the `_LibraryBuild.xml` files.) See #37651. This confusion gives me enough reasoning to do it now, I've been putting it off since 5.0.